oat-sa/extension-tao-item
# Fix/ADF-154/Move logic to tao items extension and refactor code

**Relates to:** _[ADF-154](https://oat-sa.atlassian.net/browse/ADF-154)_

**Changes:**
* `viewClassLabel` action removed;
* `oat\taoItems\model\ontology\ItemAuthorRole::INSTANCE_URI` constant deprecated;
* `oat\taoItems\model\user\TaoItemsRoles` interface with list of item roles added;
* `Item Role` user class added;
* `Item Class Navigator`, `Item Class Editor` and `Item Class Creator` roles added;
* necessary rules applied to the new roles;
* necessary action access permissions added.

**How to test:**
1. login as `super admin`;
2. create several item classes for test;
2. add user with role `Item Class Navigator`;
3. add user with role `Item Class Editor`;
4. add user with role `Item Class Creator`;
5. login as `Item Class Navigator`;
6. check access to item classes - user should be able to view item classes and their properties (without possibility to change them and create a new class);
7. login as `Item Class Editor`;
8. check access to item classes - user should be able to view item classes and edit their properties (without possibility to create a new class);
9. login as `Item Class Creator`;
10. check access to item classes - user should be able to view item classes, edit their properties and create a new class.

**Requires:**
* [oat-sa/tao-core#2883](https://github.com/oat-sa/tao-core/pull/2883)
